### PR TITLE
build alibi explain server using poetry

### PR DESCRIPTION
**Pull request by RafalSkolasinski**
**What this PR does / why we need it**:

A step towards https://github.com/SeldonIO/seldon-core/issues/2584. 
This introduces fully locked dependency for building alibi explain server image.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
```

### CODE REVIEW

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 6448e43936eb34ca1253d073da9a4a64eed22868 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3549-integration-5xhl6) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 6448e43936eb34ca1253d073da9a4a64eed22868 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3549-integration-qxkzw) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 6448e43936eb34ca1253d073da9a4a64eed22868 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3549-integration-qxkzw) | `/test integration`
notebooks | 6448e43936eb34ca1253d073da9a4a64eed22868 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-3549-notebooks-fmmdl) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 6448e43936eb34ca1253d073da9a4a64eed22868 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3549-integration-qxkzw) | `/test integration`
notebooks | 6448e43936eb34ca1253d073da9a4a64eed22868 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-3549-notebooks-wmjlq) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 6448e43936eb34ca1253d073da9a4a64eed22868 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-3549-notebooks-wmjlq) | `/test notebooks`
integration | 6448e43936eb34ca1253d073da9a4a64eed22868 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3549-integration-vr9r2) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by RafalSkolasinski** [comment]
So far integration failures:

## Run 5:
- `TestPrepack.test_mlflow_v2`

## Run 3:
- `test_rolling_deployment[graph1.json-graph2.json-True-ambas]`
- `TestPrepack.test_mlflow_v2`

**Comment by RafalSkolasinski** [comment]
image

File: components/alibi-explain-server/Dockerfile [hunk 1]
```diff
@@ -1,41 +1,65 @@
 ARG VERSION
-FROM seldonio/seldon-core-s2i-python37-ubi8:1.12.0-dev
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE}:${VERSION} as base
+
+ARG VERSION
 LABEL name="Seldon Alibi Wrapper" \
       vendor="Seldon Technologies" \
-      version="1.12.0-dev" \
+      version="$VERSION" \
```
Comment (axsaucedo) [inline components/alibi-explain-server/Dockerfile:8]: Could you validate that this won't just be replaced by the release script? I think for several docker files this is what has basically led to the version being hardcoded

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/3549#issuecomment-933273473 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [3549]: [Not Found]

**Pull request merged**